File: service-1/src/app_1/internal/configs/postgresql.py
from pydantic import Field
from pydantic_settings import BaseSettings


class PostgresqlConfig(BaseSettings):
    host: str = Field("127.0.0.1", alias="POSTGRESQL_SVC_HOST")
    port: str = Field("5432", alias="POSTGRESQL_SVC_PORT")
    database: str = Field("tech_stack_db", alias="POSTGRESQL_SVC_DATABASE")
    user: str = Field("admin", alias="POSTGRESQL_SVC_USER")
    password: str = Field("adminAdmin123!", alias="POSTGRESQL_SVC_PASSWORD")  # TODO: use SecretStr
    echo: bool = Field(True, alias="POSTGRESQL_SVC_ECHO")  # TODO: str not casted to bool


def load_postgresql_config() -> PostgresqlConfig:
    return PostgresqlConfig.model_construct(
        **PostgresqlConfig().model_dump()  # pyright: ignore [reportGeneralTypeIssues]
    )

File: migrations/src/migrations/internal/configs/postgresql.go
package configs

import (
	"log"

	"github.com/caarlos0/env"
)

type PostgresqlConfig struct {
	Host     string `env:"POSTGRESQL_SVC_HOST" envDefault:"127.0.0.1"`
	Port     string `env:"POSTGRESQL_SVC_PORT" envDefault:"5432"`
	Database string `env:"POSTGRESQL_SVC_DATABASE" envDefault:"tech_stack_db"`
	User     string `env:"POSTGRESQL_SVC_USER" envDefault:"admin"`
	Password string `env:"POSTGRESQL_SVC_PASSWORD" envDefault:"adminAdmin123!"`
}

func LoadPostgresqlConfig() PostgresqlConfig {
	config := PostgresqlConfig{}
	err := env.Parse(&config)
	if err != nil {
		log.Fatal("invalid postgresql config: ", config.Host, config.Port, config.Database, config.User)
		// panic(err)
		// panic("invalid postgresql config")
	}
	log.Println("postgresql config: ", config.Host, config.Port, config.Database, config.User)
	return config
}
